service/service_test.go

```diff
@@ -498,6 +530,219 @@ func processingOfPaymentKafkaMessageCreatesReconciliationRecords(
+
+		Convey("When the payment corresponding to the message is fetched successfully", func() {
+
+			cost := data.Cost{
+				ClassOfPayment: []string{classOfPayment},
+			}
+
+			pr := data.PaymentResponse{
+				CompanyNumber: "123456",
+				Costs:         []data.Cost{cost},
+				Refunds: []data.RefundResource{{
+					RefundId:          refundID,
+					CreatedAt:         "",
+					Amount:            0,
+					Status:            "success",
+					ExternalRefundUrl: "",
+				}},
+			}
+
+			mockPayment.EXPECT().GetPayment(paymentsAPIUrl+"/payments/"+paymentResourceID, svc.Client, apiKey).Return(pr, 200, nil).Times(1)
+
+			Convey("And the payment details corresponding to the message are fetched successfully", func() {
+
+				pdr := data.PaymentDetailsResponse{
+					PaymentStatus: "accepted",
+				}
+				mockPayment.EXPECT().GetPaymentDetails(paymentsAPIUrl+"/private/payments/"+paymentResourceID+"/payment-details", svc.Client, apiKey).Return(pdr, 200, nil).Times(1)
+
+				Convey("Then a Refund resource is constructed", func() {
+
+					refund := models.RefundResourceDao{}
+					mockTransformer.EXPECT().GetRefundResource(pr, pr.Refunds[0], paymentResourceID).Return(refund, nil).Times(1)
+
+					Convey("And committed to the DB successfully", func() {
+						mockDao.EXPECT().CreateRefundResource(&refund).DoAndReturn(func(ptr *models.RefundResourceDao) error {
+
+							// Since this is the last thing the service does, we send a signal to kill the consumer process gracefully
+							endConsumerProcess(svc, c)
+							return nil
+						})
+
+						svc.Start(wg, c)
+					})
+				})
+			})
+		})
+	})
+}
+
+// processingOfRefundKafkaMessageWithMultipleRefundsCreatesReconciliationRecords asserts that a refund of the class specified
+// will result in a call to get the payment details and the creation of refund reconciliation records with proper refund.
+func processingOfRefundKafkaMessageWithMultipleRefundsCreatesReconciliationRecords(
+	ctrl *gomock.Controller,
+	productMap *config.ProductMap,
+	classOfPayment string) {
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	c := make(chan os.Signal)
+
+	mockPayment := payment.NewMockFetcher(ctrl)
+	mockTransformer := transformer.NewMockTransformer(ctrl)
+	mockDao := dao.NewMockDAO(ctrl)
+
+	svc := createMockService(productMap, mockPayment, mockTransformer, mockDao)
+
+	Convey("Given a message is readily available for the service to consume", func() {
+
+		svc.Consumer = createMockConsumerWithRefundMessage(paymentResourceID, refundID)
+
+		Convey("When the payment corresponding to the message is fetched successfully", func() {
+
+			cost := data.Cost{
+				ClassOfPayment: []string{classOfPayment},
+			}
+
+			pr := data.PaymentResponse{
+				CompanyNumber: "123456",
+				Costs:         []data.Cost{cost},
+				Refunds: []data.RefundResource{{
+					RefundId:          refundID,
+					CreatedAt:         "",
+					Amount:            0,
+					Status:            "success",
+					ExternalRefundUrl: "",
+				},
+					{
```

Comment:
Nitpick: the formatting looks a little odd here. Have you run `go fmt ./...` at the root of the project prior to pushing?